### DISCUSSION TITLE
Is it possible to check if a record exists with two conditions?

### DISCUSSION THREAD

**Discussion by doceazedo** [pocketbase/pocketbase#1075] [Q&A] [answered]
Hi! I'm trying to create a blog post `likes` collection with the `user` and the `post` they liked. The problem is, the user should only be able to like a post once, that means, it should only exist one record with the same `user` and `post` combination.

I tried using this for my CREATE action rule, but looks like this `!()` syntax doesn't work.
```
@request.user.id != "" && !(@collection.likes.user = @request.user.id && @collection.likes.post = @request.query.post)
```

Any ideas on how I could achieve this?

**Comment by ganigeorgiev** [comment, marked as answer]
No, multi-match/exclude expressions are not possible at the moment.
Unique index/constraint across multiple fields is also not supported from the UI.

Both are in the roadmap and will be implemented in the near future.

As a workaround, for now the easiest way to handle this will be to either extend PocketBase and use the `OnRecordBeforeCreateRequest/OnRecordBeforeUpdateRequest`event hooks OR add a unique index/constraint manually on the `likes` table, eg. something like

```sql
sqlite3 pb_data/data.db "CREATE UNIQUE INDEX likes_user_post ON likes (user, post)"
```

**Reply by doceazedo** [reply]
Got it! I guess I'll use the second option for now and I'm looking forward to use it on the UI in the future. Thanks for the response and keep up the good work :)
